### PR TITLE
Add friendly output for helm package

### PR DESCRIPTION
**Pull request by FengyunPan**
1.Return err when failed to check dependencie and save chart
2.Add friendly output for helm package

### CODE REVIEW

File: cmd/helm/package.go [hunk 1]
```diff
@@ -146,7 +150,10 @@ func (p *packageCmd) run(cmd *cobra.Command, args []string) error {
 
 	name, err := chartutil.Save(ch, dest)
 	if err == nil {
-		debug("Saved %s to current directory\n", name)
+		debug("Saved into: %s\n", name)
+		fmt.Fprintf(p.out, "Saved into: %s\n", name)
```
Comment (michelleN) [inline cmd/helm/package.go]: How come you're using the same statement for debug as you are for printing out?
In this case, why would you not run the command with the `--debug` flag?
This is what the output looks like right now with the `--debug` flag:

```
helm [FengyunPan-fix-package*]$ bin/helm package docs/examples/nginx/ --debug
[debug] Saved into: /Users/michellenoorali/gocode/src/k8s.io/helm/nginx-0.1.0.tgz

Saved into: /Users/michellenoorali/gocode/src/k8s.io/helm/nginx-0.1.0.tgz
[debug] Saved /Users/michellenoorali/gocode/src/k8s.io/helm/nginx-0.1.0.tgz to /Users/michellenoorali/.helm/repository/local

Saved /Users/michellenoorali/gocode/src/k8s.io/helm/nginx-0.1.0.tgz to /Users/michellenoorali/.helm/repository/local
helm [FengyunPan-fix-package*]$
```

As you can see we're just printing out things twice now.

**Review by michelleN — CHANGES_REQUESTED** [review]

**Comment by michelleN** [comment]
@FengyunPan - I think the output statements are a little more confusing in this context. People can rely on the `--debug` flag for more verbose output. Having this output seems more confusing especially in the case of the second statement because the people packaging charts may not necessarily care or want to know about the package being saved to a local repo. That repo is only used for the `helm serve` command. I wonder if adding a statement about it here by default will confuse folks more than help them.
What do you think?

**Comment by FengyunPan** [comment]
@michelleN You are right. Thank you very much. But I think the debug info should be outputed to user,
or user can't konw what happend about 'helm package'.
What do you think?

**Comment by michelleN** [comment]
Yeah, I can understand that. How about we remove the debug line on line 110 like you've done and give a message like "Successfully packaged chart <name> and saved it to current directory"? Would that work? In the case where we save the packaged chart to a directory other than the current directory, we could output the path instead.

I think the second debug statement about saving it to the local repo should remain a debug statement because it could throw off people not using `helm serve`.

What do you think?

**Comment by FengyunPan** [comment]
@michelleN Nice! Thank you. I have updated it.

**Review by michelleN — APPROVED** [review]

**Comment by michelleN** [comment]
Thanks @FengyunPan

**Pull request merged**